src/Tribe/Utils/Post_Root_Pool.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+
+class Tribe__Utils__Post_Root_Pool {
+
+	/**
+	 * @var string
+	 */
+	protected $pool_transient_name = 'tribe_ticket_prefix_pool';
+
+	/**
+	 * @var array|bool
+	 */
+	protected static $prefix_pool = false;
+
+	/**
+	 * @var string
+	 */
+	protected $root_separator = '-';
+
+	/**
+	 * @var array
+	 */
+	protected $postfix = 1;
+
+	/**
+	 * Generates a unique root for a post using its post_name.
+	 *
+	 * @param WP_Post $post
+	 *
+	 * @return string
+	 */
+	public function generate_unique_root( WP_Post $post ) {
```

Comment:
Even better, see this:
https://github.com/moderntribe/tribe-common/pull/78/commits/d17522ad9b7289f35fe0deeaabcff1e95b5309bd